site/app/repositories/course/CourseMaterialRepository.php

```diff
@@ -13,4 +13,12 @@ public function getCourseMaterials(): array {
         $dql = 'SELECT c FROM app\entities\course\CourseMaterial c ORDER BY c.priority ASC, c.path ASC';
         return $this->_em->createQuery($dql)->getResult();
     }
+    public function findDisplayName(string $dir, string $display_name): array {
```

Comment:
Why does the display name need to be queried  ? If the display name is getting updated to the database then I think you would just need to update the entity so that whenever it gets constructed whatever will be in the DB will populate the PHP object.

Reply:
The display name gets queried when a professor wants to update the display name of a file. I wanted to make sure there wasn't another file having the same display name in the same directory. However, it doesn't work very well as for a lot of files, the display_name would be empty. This is because of two reasons: 1) files are uploaded before this change. 2) I don't update display name when uploading a file. I didn't think to implement this because it wouldn't fix #1.

Do you think it would be better to get rid of this check, add more constraints in the findDisplayName query, update display_name when new file is uploaded, perform migration...

I don't know if this will be a widely used feature (to have two different names for a file). But it does give professors the ability to rename files and move them to a different folder.

Reply:
I don't think there's a need to make sure display names are unique, they can be whatever the user wants since the filepaths are already checked to be unique.

Also why would changing the display name change the file location? If I were a user I would expect my file to be in the same location and the display name shouldn't effect that, I think that would just be confusing.


Again I think that display name should be a purely visual change that doesn't effect the real file location in anyway

Reply:
Ok. I just pushed code to remove this check. I also realized that I grab the path when users changed the display name no matter what. I changed it so path is only grabbed if the course material being edited was a link. 

Because we enforce enforce display_name (of a link) to be unique within a folder, changing the path simplifies that. 

This is the original behavior:
https://github.com/Submitty/Submitty/blob/77b1607440127cd9beac61b0e78fedfd1c24f47f/site/app/controllers/course/CourseMaterialsController.php#L392